packages/app-project/src/screens/ClassifyPage/components/WorkflowAssignmentModal/WorkflowAssignmentModalContainer.js

```diff
@@ -1,52 +1,122 @@
-import { useState, useEffect, useMemo } from 'react'
+import { useContext, useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
-import WorkflowAssignmentModal from './WorkflowAssignmentModal'
-
-function WorkflowAssignmentModalContainer ({
-  assignedWorkflowID = '',
-  currentWorkflowID = '',
-  promptAssignment = () => false
-}) {
-  const showPrompt = useMemo(() => promptAssignment(currentWorkflowID), [currentWorkflowID, promptAssignment])
-  const [ active, setActive ] = useState(showPrompt)
-  const [ dismissedForSession, setDismissed ] = useState(false)
-
-  // TODO: integrate session storage
-  function onDismiss (event) {
-    setDismissed(event.target.checked)
-  }
+import { MobXProviderContext, observer } from 'mobx-react'
+import { Button, Box, CheckBox, Text } from 'grommet'
+import { Modal, PrimaryButton, SpacedText } from '@zooniverse/react-components'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+import styled from 'styled-components'
 
-  function closeFn () {
-    setActive(false)
+import NavLink from '@shared/components/NavLink'
+
+const StyledNavLink = styled(NavLink)`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`
+
+function useStore() {
+  const { store } = useContext(MobXProviderContext)
+
+  return {
+    /** assignedWorkflowID is fetched every 5 classifications per user session */
+    assignedWorkflowID: store.user.personalization.projectPreferences.settings?.workflow_id,
+    /** This function determines if the user has an assigned workflow and verifies that workflow is active in panoptes */
+    promptAssignment: store.user.personalization.projectPreferences.promptAssignment,
   }
+}
+
+function WorkflowAssignmentModalContainer({ currentWorkflowID = '' }) {
+  const { assignedWorkflowID, promptAssignment } = useStore()
+
+  const { t } = useTranslation('screens')
+  const router = useRouter()
+  const owner = router?.query?.owner
+  const project = router?.query?.project
+  const url = `/${owner}/${project}/classify/workflow/${assignedWorkflowID}`
 
-  useEffect(() => {
-    if (showPrompt && !dismissedForSession) {
-      setActive(true)
+  /** Active state of the Modal */
+  const [active, setActive] = useState(false)
+  /** State synced with sessionStorage */
+  const [dismissedForSession, setDismissedForSession] = useState(false)
+  /** Visual/aria checkbox state */
+  const [checkboxChecked, setCheckboxChecked] = useState(false)
+
+  useEffect(function checkForDismissal() {
+    if (window.sessionStorage.getItem('workflowAssignmentModalDismissed')) {
+      setDismissedForSession(true)
+    } else {
+      setDismissedForSession(false)
     }
+  }, [])
```

Comment:
Setting state from an effect hook is an anti-pattern, and shouldn't be used. See

https://react.dev/reference/react/useState

Here, your component will always render once with the dismissed state set to `false`, then render a second time with dismissed state set to the stored value. 

To set the initial state from session storage, do something like this, which uses the stored value on the first render:

```js
const storedThing = window.sessionStorage.getItem('something');

function MyComponent() {
  const [thing, setThing] = useState(storedThing);
}
```

Reply:
Unfortunately, `window` isn't defined before the component mounts. Do you have a suggestion how to access `window` differently?

Reply:
Yeah, we’ve got a few components that render on the server but use `window` in the browser. Admin Mode is one example. 

The simplest solution is probably to leave the stored value undefined in Node. 

```js
const storedValue = sessionStorage?.getItem(‘someThing’)

```


Session storage will be undefined in Node and the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) should work in Node 14+.

Reply:
Probably worth remembering that session storage works differently from session cookies, despite the similar name. 

Did this checkbox use a session cookie in PFE? If it did, I’d use a session cookie here, and keep the name the same.

Reply:
Never mind. Looks like the original alert box didn’t have this option to turn it off. 
https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/components/workflow-assignment-dialog.jsx

Reply:
This might work better for getting initial state in the browser but not in Node:

```js
const isBrowser = typeof window === 'undefined'
const storedDismissedForSession = isBrowser && window.sessionStorage.getItem('workflowAssignmentModalDismissed')

function WorkflowAssignmentModal({ currentWorkflowID = '' }) {
  const { assignedWorkflowID, promptAssignment } = useStore()

  const { t } = useTranslation('screens')
  const router = useRouter()
  const owner = router?.query?.owner
  const project = router?.query?.project
  const url = `/${owner}/${project}/classify/workflow/${assignedWorkflowID}`

  /** Active state of the Modal */
  const [active, setActive] = useState(false)
  /** State synced with sessionStorage */
  const [dismissedForSession, setDismissedForSession] = useState(storedDismissedForSession)
  /** Visual/aria checkbox state */
  const [checkboxChecked, setCheckboxChecked] = useState(false)
```

Reply:
Writing code that runs both in Node and in the browser can be a pain when you want to work with cookies or storage.

Reply:
Thanks! That's super helpful